util/calc.py

```diff
@@ -639,7 +639,7 @@ def set_geospatial(pd_data, stream_request):
     primary_stream_len = pd_data[primary_stream.time_parameter][primary_key.as_refdes()]['data']
 
     lat_sk = stream_request.lat_stream_key
-    if pd_data.get(lat_sk.stream.time_parameter, {}).has_key(lat_sk.as_refdes()) and \
+    if lat_sk is not None and  pd_data.get(lat_sk.stream.time_parameter, {}).has_key(lat_sk.as_refdes()) and \
```

Comment:
This doesn't fix the actual problem.

If the search for the latitude/longitude stream fails in StreamRequest then stream_request.lat_stream_key will not exist. From the stack trace:

File "/home/asadev/uframes/engines/stream_engine/util/calc.py", line 637, in set_geospatial
lat_sk = stream_request.lat_stream_key
AttributeError: 'StreamRequest' object has no attribute 'lat_stream_key'

There must be an accompanying change to StreamRequest that sets these values to None in __init__